### ISSUE TITLE
Stucturing large projects

### ISSUE THREAD

**Issue by ThouCheese** [GREsau/okapi#30]
Hi @GREsau,

I'm using okapi in a large project (~200 endpoints), and all of these endpoints need to be documented. Currently, I have separated these endpoints into modules, that each publicly export a function called `get_routes`. This function internally uses the `routes_with_openapi` macro to generate a `Vec<rocket::Route>`, and then the Rocket instance mounts the results of these functions on a separate mount point. The issue with this approach is that this will generate many `openapi.json` files instead of one. This in turn does not play well with `swagger-codegen` to automatically generate client libraries, and it is unfamiliar for many front enders. In the ideal case, I think we should generate one `openapi.json` file containing information about all documented routes in the Rocket instance.

I talked to @ralpha about this, and we both tried to cobble together something basic, but both our approaches hit a dead end. My approach was to create two macros called `merge_routes` and `mount_routes`, which supersedes the mounting functionality in Rocket. `mount_routes` would handle the namespacing that is normally handled by Rocket, and the merge routes would merge together the `openapi.json`'s that have been created for each group of routes. I had imagined to build something along these lines:
```rust
let routes = merge_routes![
    mount_routes!["/module1", routes_with_openapi![route1, route2]],
    mount_routes!["/module2", routes_with_openapi![route3, route4]],
];
rocket::ignite().mount("/", routes).launch();
```
I seems a little silly to me however, to redo the work that Rocket is already doing, and to merge the json files instead of maybe generating one already correct file. I was wondering I you had run into this problem yet, and if so, if you had any thoughts on how to solve it?

**Comment by ralpha** [comment]
I added the macro `rocket_okapi::mount_endpoints_and_merged_docs` in order to solve this problem.
Will be fixed by next release.

**Comment by ThouCheese** [comment]
Nice! Will it be released together with Rocket 0.5?

**Comment by ralpha** [comment]
> Nice! Will it be released together with Rocket 0.5?

I don't know when Rocket 0.5 is going to be released.
But right now okapi/rocket_okapi is in its final state for the next release on crates.io . (if no bugs come up)
Just waiting for GREsau to review it is create a new release.
